tools/cpp/unix_cc_configure.bzl

```diff
@@ -84,6 +84,21 @@ def _escaped_cplus_include_paths(repository_ctx):
   else:
     return []
 
+def _linker_flags(repository_ctx):
+  """Use ${LINK_FLAGS} to compute the list of flags for linkflag."""
+  if "LINK_FLAGS" not in repository_ctx.os.environ:
+    return [
+      "-lstdc++",
+      "-lm",  # Some systems expect -lm in addition to -lstdc++
+    ]
+
+  result = []
+  # FIXME: For paths : is already narrow but for flags it is 
+  # probably too narrow. Needs to replaced by way of properly
+  # splitting including escaping.
+  for p in repository_ctx.os.environ["LINK_FLAGS"].split(":"):
```

Comment:
Why did yo choose ':' for separator? Is that what cmake uses? Which flags wouldn't work with naive split-on-space approach? Linker flags can be passed down as `-Wl,-foo,bar,baz` for `-foo bar baz`, right?

Also what is your reasoning behind naming these variables? Why not BAZEL_LINKER_FLAGS for consistency with your other pr?

And finally, could you add this env var to cc_configure.bzl? Thanks!

Reply:
> Why did yo choose ':' for separator?

No particular reason. It is a bad choice though!

> Is that what cmake uses?

CMAKE uses `;`. Which is a bit more sane but still bad.

> Linker flags can be passed down as -Wl,-foo,bar,baz for -foo bar baz, right?
It _should_ work AFAIK. Seems very hackish to me. Someone with more experience might know - summoning @meteorcloudy

> Which flags wouldn't work with naive split-on-space approach?
Stuff like `-l :foo`. I vagely remember that there are some special use cases like this in gcc/ld.

> Why not BAZEL_LINKER_FLAGS for consistency with your other pr?
Well from the current set of environment variable names it is not clear to me which model they follow. +1 for `BAZEL_LINKER_FLAGS`, though.

Reply:
I don't think `-Wl,-foo,bar,baz` is hackish, I see it used very often, internally and externally. I wouldn't worry about it. You still can express versioned libraries with `-Wl,-l,:foo` if you need to.

Yeah there's not much consistency there. BAZEL_* used to mean it's needed for bootstrapping bazel, but now it's also used to tweak windows toolchain. 

Anyway, bonus points for adding comment to cc_configure.bzl for what this variable does :)

Reply:
> Linker flags can be passed down as -Wl,-foo,bar,baz for -foo bar baz, right?

The `-Wl,xxx` option for gcc passes a comma-separated list of tokens as a space-separated list of arguments to the linker. So
`gcc -Wl,aaa,bbb,ccc`
eventually becomes a linker call
`ld aaa bbb ccc`

So if this "linker_flag"s are passed to gcc, we should add `-Wl`

Reply:
@meteorcloudy There is an awful lot of `gcc` in these sentences. I am using clang currently (which AFAIK supports a lot of strange gcc behavior) and others even stranger compilers. Are we sure the `-Wl` "variant" works for allCompilersThereAre?

Reply:
IIRC, like Marcel said, we are often using this syntax to pass linker flags, so I believe it should be no problem. Marcel might know more about clang.

Reply:
Clang is the same, that's the contract there is for passing flags from compiler driver down to the linker. I have no idea what msvc does, but also msvc toolchain autoconfiguration is not as tweakable as linux/mac toolchain, so I wouldn't worry too much about it.